Libraries/src/Amazon.Lambda.RuntimeSupport/Helpers/ConsoleLoggerWriter.cs

```diff
@@ -342,6 +442,116 @@ private string ConvertLogLevelToLabel(LogLevel level)
                 return level.ToString();
             }
 
+            private string ConvertToJsonFormattedMessage<TEntry>(string displayLevel, TEntry entry)
+            {
+                var bufferWriter = new ArrayBufferWriter<byte>();
+                using var writer = new Utf8JsonWriter(bufferWriter, new JsonWriterOptions
+                {
+                    Indented = false
+                });
+
+                writer.WriteStartObject();      // root json object
+
+                writer.WriteString("Timestamp", DateTime.UtcNow.ToString(_dateFormat, CultureInfo.InvariantCulture));
+
+                writer.WriteString("AwsRequestId", CurrentAwsRequestId);    // allows log message to be discoverable by X-ray
+
+                writer.WriteString("Level", displayLevel);
+
+                writer.WriteString("Message", entry.ToString());
+
+                if (entry is IMessageEntry messageEntry)
```

Comment:
Going to be another one of those challenging parts where you can't have RuntimeSupport use a new type on Lambda Core. Probably have to do something where the message entry also implement the `IDictionary` interface or something.

Reply:
To work around this I've changed the implementation of `FormattedMessageEntry`. It is a `IReadOnlyList<KeyValuePair<string, object>>` with the first entry (index 0) being `["{Exception}"] = <log_exception>`. The rest of the entries in that list are log name-value parameters. The log writer will recognize this magic string and renders the exception and state. This is the only way I can think of to avoid introducing new types in `Amazon.Lambda.Core`.